build-aux/snap/snapcraft.yaml

```diff
@@ -29,9 +28,36 @@ passthrough:
 # https://forum.snapcraft.io/t/5547/10
 parts:
   snapd:
-    # FIXME: this should probably go upstream
-    plugin: x-builddeb
+    plugin: nil
     source: .
+    build-snaps: [go/1.10/stable]
+    override-pull: |
+      snapcraftctl pull
+      # install build dependencies
+      export DEBIAN_FRONTEND=noninteractive
+      export DEBCONF_NONINTERACTIVE_SEEN=true
+      sudo apt-get build-dep -y ./
+      ./get-deps.sh --skip-unused-check
+      # set version after installing dependencies so we have all the tools here
+      snapcraftctl set-version $(./mkversion.sh --output-only)
+    override-build: |
+      # unset the LD_FLAGS and LD_LIBRARY_PATH vars that snapcraft sets for us
+      # as those will point to the $SNAPCRAFT_STAGE which on re-builds will 
+      # contain things like libc and friends that confuse the debian package
+      # build system
+      # TODO: should we unset $PATH to not include $SNAPCRAFT_STAGE too?
+      unset LD_FLAGS
+      unset LD_LIBRARY_PATH
+      # if we are root, disable tests
```

Comment:
Saying why we disable tests would be more useful. I see what the code does but am not the wiser.

Reply:
sure I'll add this in a followup